### PR TITLE
docs: create ERRORS.md for x/module

### PR DESCRIPTION
**Pull request by 170210**
## Description
This PR has following changes.

- create an automatic error document generation module
- check generated error documents in CI workflow

closes: #942 

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Comment by ulbqb** [comment]
Please add go.mod in tools/error_doc.

**Comment by ulbqb** [comment]
How about adding link to x/ERRORS.md in x/README.md?

File: tools/error_doc/main.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+	// get all errors.go in x folder
+	errorFile := "errors.go"
+	filePaths, err := findFilesWithName(targetPath, errorFile)
+	if len(filePaths) == 0 || err != nil {
+		return nil, nil, errors.New("not find target files in x folder")
+	}
+
+	// get each module name and bind it to paths (one module may have multiple errors.go)
+	moduleWithPaths := make(map[string][]string)
```
[the remark this answers is not in the document]
Comment (170210) [reply tools/error_doc/main.go:43]: Yes, in the token module, there are two errors.go file.
[token/class/errors.go](https://github.com/170210/finschia-sdk/blob/feature/issue_942/x/token/class/errors.go)
[token/errors.go](https://github.com/170210/finschia-sdk/blob/feature/issue_942/x/token/errors.go)

**Review by ulbqb — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -58,3 +58,4 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ### Document Updates
 * (readme) [\#997](https://github.com/finschia/finschia-sdk/pull/997) fix swagger url
+* (x/ERRORS.md) [\#1059](https://github.com/Finschia/finschia-sdk/pull/1059) create ERRORS.md for x/module
```
Comment (zemyblue) [inline CHANGELOG.md:61]: Please update prefix word from `x/ERROR.md` to `docs`.

**Review by zemyblue — APPROVED** [review]

**Review by ulbqb — DISMISSED** [review]

**Review by tnasu — COMMENTED** [review]
How about renaming the file to `error_docs_generator.go`?

**Review by ulbqb — DISMISSED** [review]
LGTM

File: tools/error_doc/generator/error_docs_generator.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+			filepath:  filepath,
+			codespace: "",
+			constDict: make(map[string]string),
+			errorDict: []errorInfo{},
+		})
+	}
+	// sort keys and filepaths
+	for moduleName := range edg.errorDocument {
+		edg.modules = append(edg.modules, moduleName)
+		// sort.Strings(edg.errorDocument[moduleName])
```
Comment (tnasu) [inline tools/error_doc/generator/error_docs_generator.go:73]: Would you remove this?

**Review by zemyblue — APPROVED** [review]

**Review by ulbqb — APPROVED** [review]

**Pull request merged**